apps/web/src/pages/UpcomingLessons.tsx

```diff
@@ -56,6 +57,14 @@ const UpcomingLessons: React.FC = () => {
     canceled: true,
   });
 
+  // used to show rate dialog only if the lesson is already started
+  const ratingLessonSelfQuery = useFindLesson(rateDialogInfo.lessonId || 0);
+  const isRatingLessonStarted = useMemo(() => {
+    return dayjs
+      .utc(ratingLessonSelfQuery.data?.lesson.start)
+      .isBefore(dayjs.utc());
+  }, [ratingLessonSelfQuery.data]);
+
```

Comment:
Instead if sending a request for the backend to get the lesson info. You can access the lesson start time from the url.